ebean-core/src/main/java/io/ebeaninternal/server/querydefn/MdBindHash.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed Materials - Property of FOCONIS AG
+ * (C) Copyright FOCONIS AG.
+ */
+
+package io.ebeaninternal.server.querydefn;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.UUID;
+
+import io.ebeaninternal.api.BindHash;
+
+/**
+ * Bind hash that uses a MessageDigest to compute a collision resistent hash.
+ *
+ * @author Roland Praml, FOCONIS AG
+ *
+ */
+public class MdBindHash implements BindHash {
+  private MessageDigest md;
+  private byte[] buffer;
+  private int hashCode;
+
+  public MdBindHash(String algorithm) {
+    try {
+      md = MessageDigest.getInstance(algorithm);
+    } catch (NoSuchAlgorithmException nsae) {
+      throw new InternalError(algorithm + " not supported", nsae);
+    }
+  }
+
+  @Override
+  public BindHash update(int v) {
+    md.update((byte) (v >>> 24));
+    md.update((byte) (v >>> 16));
+    md.update((byte) (v >>> 8));
+    md.update((byte) (v >>> 0));
+    return this;
+  }
+
+  @Override
+  public BindHash update(long v) {
+    md.update((byte) (v >>> 56));
+    md.update((byte) (v >>> 48));
+    md.update((byte) (v >>> 40));
+    md.update((byte) (v >>> 32));
+    md.update((byte) (v >>> 24));
+    md.update((byte) (v >>> 16));
+    md.update((byte) (v >>> 8));
+    md.update((byte) (v >>> 0));
+    return this;
+  }
+
+  @Override
+  public BindHash update(boolean boolValue) {
+    md.update(boolValue ? (byte) 1 : (byte) 0);
+    return this;
+  }
+
+  @Override
+  public BindHash update(Object value) {
+    if (value == null) {
+      md.update((byte) 0);
+
+      // do some special handling for known object types
```

Comment:
I think it makes sense, to use some special handling for certain classes
(these are the datatypes we mainly use in our application)
for all others, fall back to hashCode. This class can be easily extended to add other datatypes

@rbygrave maybe I make it in a way configurable, that you can specify the className in the ServerConfig or load it with ServiceLoader. What do you think?